### DISCUSSION TITLE
api

### DISCUSSION THREAD

**Discussion by kajvans** [louislam/dockge#459] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API

### 🔖 Feature description

there should be an api so you could see stats in dashboard

### ✔️ Solution

a route you can call where you can see the stats that are displaying normally on the dashboard page

**Comment by retnuh-code** [comment]
Bump!

Something similar to the Portainer API, trying to hook up Homepage currently.

**Comment by Pr0mises** [comment]
The API could also include updating a specific stack like portainer does (PUSH Request towards dockge.lan/<some random number OR stackname>

**Reply by vdaron** [reply]
This is my most wanted feature, to be able to integrate dockge with Continuous Integration

Thanks !

**Comment by drwahl** [comment]
I would like to have something that is Swagger/OpenAPI compliant, too, so that tools that generate clients from OpenAPI spec could be used. A quick search suggests [vue-openapi](https://github.com/koumoul-dev/vue-openapi) should make this relatively simple.

**Comment by giannis371** [comment]
+1
